src/Sarif.Sarifer/BackgroundAnalysisService.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.Sarif.Sarifer
+{
+    /// <summary>
+    /// Performs static analysis in the background.
+    /// </summary>
+    // TODO: Reanalyze when buffer changes.
+    // TODO: Remove error list items when buffer closes.
+    [Export(typeof(IBackgroundAnalysisService))]
+    internal class BackgroundAnalysisService : IBackgroundAnalysisService
+    {
+#pragma warning disable CS0649 // Filled in by MEF
+#pragma warning disable IDE0044 // Assigned by MEF
+        [ImportMany]
+        private IEnumerable<IBackgroundAnalyzer> analyzers;
+#pragma warning restore IDE0044
+#pragma warning restore CS0649
+
+        /// <inheritdoc/>
+        public void StartAnalysis(string path, string text)
+        {
+            if (this.analyzers.Any() == true)
+            {
+                foreach (IBackgroundAnalyzer analyzer in this.analyzers)
+                {
+                    analyzer.StartAnalysis(path, text);
```

Comment:
>StartAnalysis [](start = 29, length = 13)

I would think that would be StartAnalysisAsync and that this class would keep an array of running analysis tasks. (should likely also integrate with VS task center with some sort of progress so the user knows something is happening in the background)

Reply:
Yes, we do need to keep track of what's running and what's been done. I propose to accept the small/easy fixes in this PR, and then we'll use it as the basis for future work. Much of that work will be done by @eddynaka, per @michaelcfanning.

---
In reply to: [519919012](https://github.com/microsoft/sarif-visualstudio-extension/pull/260#discussion_r519919012) [](ancestors = 519919012)